tdrs-backend/tdpservice/search_indexes/reparse.py

```diff
@@ -0,0 +1,110 @@
+"""Reparsing command for selected files."""
+# should include all the steps in the management command
+from tdpservice.data_files.models import DataFile
+from tdpservice.search_indexes.models.reparse_meta import ReparseMeta
+from tdpservice.core.utils import log
+from tdpservice.users.models import User
+from tdpservice.search_indexes.utils import (
+    backup,
+    get_log_context,
+    assert_sequential_execution,
+    should_exit,
+    handle_elastic,
+    delete_associated_models,
+    count_total_num_records,
+    calculate_timeout,
+    handle_datafiles,
+)
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def clean_reparse(selected_file_ids):
+    """Reparse selected files."""
+    selected_files = [int(file_id) for file_id in selected_file_ids[0].split(",")]
+
+    ######
+    files = DataFile.objects.filter(id__in=selected_files)
+    backup_file_name = "/tmp/reparsing_backup"
+    backup_file_name += "_selected_files"
+    continue_msg = "You have selected to reparse datafiles for FY {fy} and {q}. The reparsed files "
+    continue_msg = continue_msg.format(
+        fy=f"selected files: {str(selected_files)}", q="Q1-4"
+    )
+
+    num_files = files.count()
+
+    # add fmt_str
+
+    system_user, created = User.objects.get_or_create(username="system")
+    if created:
+        logger.info("Created system user")
+    log_context = get_log_context(system_user)
+
+    all_fy = "All"
+    all_q = "Q1-4"
+
+    log(
+        f"Starting clean_and_reparse for {num_files} files",
+        logger_context=log_context,
+        level=logging.INFO,
+    )
+
+    is_sequential = assert_sequential_execution(log_context)
+    should_exit(not is_sequential)
```

Comment:
Should we update this to raise an exception/return an error to the user so that they know why the reparse didn't happen? Even writing to the console would be a start.

Reply:
Also, can we move this to the very beginning of the function to avoid unnecessary computation if we arent sequential?